chesster.js

```diff
@@ -62,268 +47,89 @@ function critical_path(todo){
-                    bot.reply(message, prepareCaptainsMessage(self.teams));
-                });
-            });
-        });
-    });
-});
-
-function getCaptains(self, callback){
-    self.sheet.getCells({
-        "min-row": TEAM_START_ROW,
-        "max-row": TEAM_END_ROW,
-        "min-col": BOARD_1_NAME, 
-        "max-col": BOARD_6_NAME,
-	"return-empty": true,
-    }, function(err, cells) {
-        var num_cells = cells.length;
-        for(var ci in cells){
-            var cell = cells[ci];
-            var team_index = cell.row - TEAM_START_ROW; // number of rows before team name start
-            switch(cell.col){
-                case BOARD_1_NAME:
-                case BOARD_2_NAME:
-                case BOARD_3_NAME:
-                case BOARD_4_NAME:
-                case BOARD_5_NAME:
-                case BOARD_6_NAME:
-                if(cell.value.includes("*")){
-                    if(!self.teams[team_index]){
-                        self.teams[team_index] = {};
-                    }
-                    self.teams[team_index]["captain"] = cell.value.replace("*", "");
-                }
-                break;
-            }
-        }
-        callback();
-    });
-}
-
-function prepareCaptainsMessage(teams){
-    var message = "Team Captains:\n";
-    var teamIndex = 1;
-    teams.forEach(function(team, index, array){
-        message += "\t" + (teamIndex++) + ". " + team.name + ": " + (team.captain || "Unchosen") + "\n";
-    });
-    return message;
-}
-
-function sayCaptains(self, convo, callback){
-    convo.say(prepareCaptainsMessage(self.teams));
-    convo.next();
-}
 
 /* rating */
 
-controller.hears([
-    players.appendPlayerRegex("rating", true)	
-],[
-	'direct_mention', 
-	'direct_message'
-],function(bot,message) {
-    bot_exception_handler(bot, message, function(){
-        var player_name = players.getSlackUser(users, message).name;
-        getRating(player_name, function(rating){
-            if(rating){
-                bot.reply(message, prepareRatingMessage(player_name, rating));
-            }else{
-                bot.reply(message, "I am sorry. I could not find that player.");
-            }
-        });
-    });
-});
-
-function getRating(player, callback){
-    getPlayerByName(player, function(error, opponent){
-        if(opponent){
-            getClassicalRating(opponent, callback);
+chesster.hears({
+    patterns: [players.appendPlayerRegex("rating", true)],
+    messageTypes: [
+        'direct_mention', 
+        'direct_message'
+    ]
+},
+function(bot,message) {
+    var playerName = players.getSlackUser(users, message).name;
+    return lichess.getPlayerRating(playerName).then(function(rating) {
+        if(rating){
+            bot.reply(message, prepareRatingMessage(playerName, rating));
         }else{
-            console.error(JSON.stringify(error));
-            callback();
+            bot.reply(message, "I am sorry. I could not find that player.");
```

Comment:
instead of implementing this as an else clause of the conditional, could we use the second argument of the `then(...)` function to send the response? 

Not sure if maybe the second argument is reserved for other kinds of errors or if this is a valid use-case.